docs/concepts/node.rst

```diff
@@ -0,0 +1,90 @@
+====
+Node
+====
+
+The Node is the basic unit of Hamilton. Each node contains code to execute, and they can be assembled to form a dataflow.
+
+Nodes are simply Python functions
+---------------------------------
+Nodes are defined using the native Python function definition API. The figure below shows how the functions ``A()``, ``B()``, and ``C()`` are automatically assembled into a dataflow. Notice that function definitions don't have any special "Hamilton code". Also, the function name and return type ``A() -> int`` match the parameter ``A: int`` found in functions ``B()`` and ``C()``.
```

Comment:
I don't think this is the way to introduce "node". We start on all the other pages talking about functions, so we should start with that here and transition to node -- or do we even need to talk about nodes?

Reply:
it's true that the welcome page should be adjusted accordingly